app/models/user.rb

```diff
@@ -1660,6 +1660,14 @@ def disable_redesigned_user_menu
     Discourse.redis.del("#{REDESIGN_USER_MENU_REDIS_KEY_PREFIX}#{self.id}")
   end
 
+  def sidebar_categories_ids
+    return category_sidebar_section_links.pluck(:linkable_id) if category_sidebar_section_links.present?
+    if SiteSetting.default_sidebar_categories.present?
+      return Category.secured(guardian).where(id: SiteSetting.default_sidebar_categories.split("|").map(&:to_i)).pluck(:id)
```

Comment:
The `CategoryGuardian` has a `allowed_category_ids` method. I wonder if we can pass the guardian to this method and just ensure the site setting category ids are present in the array instead of executing another query.